docs/source/jit_language_reference_statements.rst

```diff
@@ -0,0 +1,389 @@
+
+::
+    assignment_stmt ::=  (target_list "=")+ (starred_expression)
+    target_list     ::=  target ("," target)* [","]
+    target          ::=  identifier
+                     | "(" [target_list] ")"
+                     | "[" [target_list] "]"
+                     | attributeref
+                     | subscription
+                     | slicing
+                     | "*" target
+    `target <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-target>`
+    `identifier <https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-identifier>`
+    `attributeref <https://docs.python.org/3/reference/expressions.html#grammar-token-attributeref>`
+    `subscription <https://docs.python.org/3/reference/expressions.html#grammar-token-subscription>`
+    `slicing <https://docs.python.org/3/reference/expressions.html#grammar-token-slicing>`
+    `target_list <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-target-list>`
+    `starred_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-starred-expression>`
+
+* `yield expression` is not supported in Torchscript.
+
+Augmented assignment statements:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    augmented_assignment_stmt ::=  augtarget augop (expression_list)
+    augtarget ::=  identifier | attributeref | subscription
+    augop ::=  "+=" | "-=" | "*=" | "/=" | "//=" | "%=" |
+                "**="| ">>=" | "<<=" | "&=" | "^=" | "|="
+    `augtarget <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-augtarget>`
+    `augop <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-augop>`
+    `identifier <https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-identifier>`
+    `attributeref <https://docs.python.org/3/reference/expressions.html#grammar-token-attributeref>`
+    `subscription <https://docs.python.org/3/reference/expressions.html#grammar-token-subscription>`
+    `expression_list <https://docs.python.org/3/reference/expressions.html#grammar-token-expression-list>`
+
+* `yield expression` is not supported in Torchscript.
+
+Annotated assignment statements:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    annotated_assignment_stmt ::=  augtarget ":" expression
+                               ["=" (starred_expression)]
+    `augtarget <https://docs.python.org/3/reference/simple_stmts.html#grammar-token-augtarget>`
+    `expression <https://docs.python.org/3/reference/expressions.html#grammar-token-expression>`
+    `starred_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-starred-expression>`
+
+* `yield expression` is not supported in Torchscript.
+
+The ``raise`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    raise_stmt ::=  "raise" [expression ["from" expression]]
+    `expression <https://docs.python.org/3/reference/expressions.html#grammar-token-expression>`
+
+* Raise statements in TorchScript do not support ``try\finally``.
```

Comment:
I think `try/except/finally` should be discussed separately. It is slightly strange to say "raise" doesn't support "try".

Reply:
What do you mean discussed separately? They are not supported in TS?